frontend/catalyst/jax_tracer.py

```diff
@@ -208,6 +208,44 @@ def _eval_jaxpr(*args, **kwargs):
 }
 
 
+@debug_logger
+def uses_transform(qnode, transform_name, mode: Literal["only_one", "any"] = "any"):
+    """
+    Detect if a QNode uses specific transform that is specified by `transform_name`.
+
+    Args:
+        qnode: The quantum node to check
+        transform_name: Name of the transform to look for
+        mode: If "only_one", returns True only if transform_name is the ONLY transform
+                 in the program. If "any", returns True if transform_name is present in the program.
+
+    Returns:
+        bool: True if `transform_name` is detected (and is only one if only_one=True),
+              False otherwise
+    """
+    if not (hasattr(qnode, "transform_program") and qnode.transform_program):
+        return False
+
+    transform_funcs = [
+        transform_container.transform
+        for transform_container in qnode.transform_program
+        if hasattr(transform_container, "transform")
+    ]
+
+    matching_transforms = [
+        func
+        for func in transform_funcs
+        if hasattr(func, "__name__") and transform_name in func.__name__
+    ]
+
+    has_target_transform = len(matching_transforms) > 0
+
+    if mode == "only_one":
+        return has_target_transform and len(transform_funcs) == 1
+
+    return has_target_transform
```

Comment:
Since this helper only counts, we don't need to copy things into and out of new lists!

```suggestion
   num_matching_transforms = 0
   for func in transform_funcs:
      if transform_name in func.__name__:
          num_matching_transforms += 1
    
    ...
```

Also, why is it `transform_name in func.__name__`? Is it only a substring?